README.md

```diff
@@ -53,6 +53,30 @@ See [lock-filename](#lock-filename) for more information on changing the default
 lockfile paths and [pip-compile-constraint](#pip-compile-constraint) for more
 information on syncing dependency versions across environments.
 
+### Updating Dependencies
+
+Passing arguments like `--upgrade` / `--upgrade-package` to `pip-compile` is
+controlled by two environment variables: `PIP_COMPILE_UPGRADE` and
+`PIP_COMPILE_UPGRADE_PACKAGE`. When either of these environment variables are
+set `hatch` will assume that the environment is out of date and automatically
+run `pip-compile` the next time the environment is activated.
+
+The below command runs `pip-compile --upgrade` on the `default` environment.
+
+```shell
+PIP_COMPILE_UPGRADE=1 hatch env run --env default -- python --version
```

Comment:
Honestly I still think that deleting the lock file is the more ergonomic usage :/

Reply:
Fair enough. I mentioned that on the latest update. I think the updated README does a decent job of explaining the upgrade process.